observability/interceptor.go

```diff
@@ -40,7 +44,7 @@ func (i *Interceptor) WrapUnary(next connect.UnaryFunc) connect.UnaryFunc {
 		resp, err := next(ctx, req)
 
 		if verb := req.Header().Get(headers.VerbHeader); verb != "" {
```

Comment:
This is not correct - the header contains every hop in the call chain, so this will always be the originating call, not the caller.

Reply:
You'll have to use headers.Callers() then extract the last two entries to get the caller and callee.